src/com/sap/piper/ConfigurationHelper.groovy

```diff
@@ -96,13 +96,17 @@ class ConfigurationHelper implements Serializable {
         return this
     }
 
-    Map use(){ return config }
+    Map use(){
+        MapUtils.traverse(config, { v -> (v instanceof GString) ? v.toString() : v })
+        return config
+    }
 
     ConfigurationHelper(Map config = [:]){
         this.config = config
     }
 
     def getConfigProperty(key) {
+        use()
```

Comment:
[side note] This is a good example for a code smell introduced since we try to solve more than one aspect in one class. I recall #210: Normally there would be one call to ```use``` when we are finished with setting up the configuration. Now we re-use ```use``` here in order to ensure we are in a proper state. In case we retrieve several properties by ```getConfigProperty``` we call use several times. Now it would be possible to introduce a boolean flag telling us 'use was already called', but that makes things even worse, since in this case we would have to update the flag, e.g. in case some more parameters are mixed in ...

Reply:
if **_withMandatoryProperty()_** is a function which belongs to the builder pattern, it should be decoupled from **_getConfigProperty()_** to avoid unnecessary traversals.
e,g, the step **_transportRequestUploadFile_** calls _**withMandatoryProperty()**_  almost 10 times.

Reply:
I would like to propose another way: Make ```getConfigProperty``` private. In this case there is no need to call ```use``` from ```getConfigProperty```. From the perspective of the ```withMandatoryParameters``` method there is no need to have any specific types, it is only important if there is a value or not. So there is in general no need to call ```use``` before checking if there is a particular parameter defined.

Last year we decided that everything inside the ```src``` folder is not an API. From that point of view it is easy to change that. But we also know that our Potsdam colleagues uses that class from another shared lib. On the one hand that violates the approach 'everything in src not an API' on the other hand we would - of course - like to be polite.